apina-gradle/src/test/kotlin/fi/evident/apina/gradle/ApinaIntegrationTest.kt

```diff
@@ -59,6 +59,11 @@ class ApinaIntegrationTest {
                     throw new UnsupportedOperationException();
                 }
 
+                @GetMapping("/cat")
+                public ParameterizedResponseType<Cat, Dog> cat(@RequestBody ParameterizedRequestType<Elephant> foo) {
```

Comment:
Great to have this integration test, but it would be nice to have a unit test for as well. Those are the most important tests for Apina. The integration test mostly checks that nothing breaks with regarding to Gradle API. (Well, even the "unit tests" end up testing integration between several components, but they are still closer to the actual code than the integration tests.)

Perhaps something like this in `JacksonTypeTranslatorTest`:

```kotlin
@Test
@Suppress("unused")
fun `generic type parameters are translated for unknown generic types`() {
    class Foo
    class Bar
    class GenericType<T, S>

    class Root {
        lateinit var foo: GenericType<Foo, Bar>
    }

    val model = JavaModel().apply {
        loadClassesFromInheritanceTree<Root>()
        loadClassesFromInheritanceTree<GenericType<*, *>>()
        loadClassesFromInheritanceTree<Foo>()
        loadClassesFromInheritanceTree<Bar>()
    }
    val api = ApiDefinition()
    val translator = JacksonTypeTranslator(settings, model, api)
    translator.translateType(JavaType.basic<Root>(), MockAnnotatedElement(), TypeEnvironment.empty())

    assertTrue(api.classDefinitions.any { it.type.name == "Foo" }, "Class definition for Foo is created")
    assertTrue(api.classDefinitions.any { it.type.name == "Bar" }, "Class definition for Bar is created")
}
```

Reply:
Yes, you are totally right, I just changed the integration test because it was a lot easier for me, but having that unit test is a much better idea. 

In fact I think it´s better if you get rid of my changes in the integration test to keep the consistency (no sense to have dogs, cats and elephants there :D)